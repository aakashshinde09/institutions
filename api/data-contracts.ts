/* eslint-disable */
/* tslint:disable */
/*
 * ---------------------------------------------------------------
 * ## THIS FILE WAS GENERATED VIA SWAGGER-TYPESCRIPT-API        ##
 * ##                                                           ##
 * ## AUTHOR: acacode                                           ##
 * ## SOURCE: https://github.com/acacode/swagger-typescript-api ##
 * ---------------------------------------------------------------
 */

export interface NGOLookup {
  id: number
  graphId: string
  name: string
  imageLink: string
  description: string
  link: string
  rewardsOwner: string
  ngoAddress: string
  totalStacked: number
  totalDonation: number
  lastDonation?: string
  donationPercent?: number
  totalUserStaked: number
  location: string
  balance: number
  stakes?: Stakes[]
}

export interface Stakes {
  balance: number
  createdAt: string
  customStakeId: number
  stackedAmount: number
  percentShare: number
  totalDonation: number
  startDate: string
}

export type ApiResponse = object

export interface RewardLookup {
  id: number
  stackedAmount: number
  balanceWithoutProfit: number
  profit: string
  donation: string
  ngoAddress: string
  /** @format date-time */
  createdAt: string
  serviceFee: string
  stake?: Stakes
}

export interface AuthTokensDto {
  accessToken: string
  refreshToken: string
}

export interface SignDto {
  publicAddress: string
}

export interface AprDto {
  Apr: number
  Timestamp: number
}

export interface AverageAprDto {
  Aprs: AprDto[]
  SmaArp: number
  Symbol: string
  Address: string
  ChainId: number
}

export interface TransactionLookup {
  id: number
  hash: string
  transactionType: string
  fromAddress: string
  toAddress: string
  ethAmount: number
  networkFee: number
  blockNumber: number
  blockTimestamp: number
}

export interface WithdrawRequestLookup {
  id: number
  graphId: string
  customRequestId: number
  ngoAddress: string
  status: 1 | 2
  stakerAddress: string
  createdAt: string
  stake: Stakes
}

export interface WithdrawStEthRequestLookup {
  id: string
  ngo: string
  claimer: string
  stake: Stakes
  blockTimestamp: number
}

export interface NgoControllerGetAllParams {
  take: number
  skip: number
}

export type NgoControllerGetAllData = ApiResponse & {
  data?: {
    items?: NGOLookup[]
    /** @default 1 */
    totalItems?: number
  }
  /** @default 200 */
  status?: number
  error?: string
}

export type NgoControllerPullNgoFromGraphData = any

export type NgoControllerDeleteNgoByIdData = any

export interface RewardControllerGetCurrentUserRewardsParams {
  take: number
  skip: number
}

export type RewardControllerGetCurrentUserRewardsData = ApiResponse & {
  data?: {
    items?: RewardLookup[]
    /** @default 1 */
    totalItems?: number
  }
  /** @default 200 */
  status?: number
  error?: string
}

export type RewardControllerPullRewardFromGraphAndScData = any

export type RewardControllerDeleteAllData = any

export type RewardControllerDeleteOneData = any

export type UserControllerGeCurrentUserDepositsInformationData = any

export type UserControllerGetCurrentUserNgoListData = ApiResponse & {
  data: NGOLookup[]
  /** @default 200 */
  status?: number
  error?: string
}

export type AuthControllerSignInData = ApiResponse & {
  data?: AuthTokensDto
  /** @default 200 */
  status?: number
  error?: string
}

export type AuthControllerSignUpData = any

export interface AuthControllerRefreshPayload {
  refreshToken?: string
}

export type AuthControllerRefreshData = ApiResponse & {
  data?: AuthTokensDto
  /** @default 200 */
  status?: number
  error?: string
}

export type AprControllerGetAverageAprData = ApiResponse & {
  data?: AverageAprDto
  /** @default 200 */
  status?: number
  error?: string
}

export interface TransactionControllerGetByWalletAddressParams {
  take: number
  skip: number
}

export type TransactionControllerGetByWalletAddressData = ApiResponse & {
  data?: {
    items?: TransactionLookup[]
    /** @default 1 */
    totalItems?: number
  }
  /** @default 200 */
  status?: number
  error?: string
}

export interface WithdrawControllerGetByWalletAddressParams {
  take: number
  skip: number
  status: 1 | 2
}

export interface withdrawControllerGetWithdrawStEth {
  take: number
  skip: number
  claimerAddress: string
}

export type WithdrawControllerGetByWalletAddressData<T = WithdrawRequestLookup> =
  ApiResponse & {
    data?: {
      items?: T[]
      /** @default 1 */
      totalItems?: number
    }
    /** @default 200 */
    status?: number
    error?: string
  }
