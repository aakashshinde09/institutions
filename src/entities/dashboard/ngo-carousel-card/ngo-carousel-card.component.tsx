import type { Stakes } from '@api/data-contracts'
import { Tooltip } from '@mantine/core'
import { formatNumber } from '@shared/lib/utils'
import { CaptionItem } from '@shared/ui/caption-item'

interface INgoCarouselCardProperties {
  data?: Stakes
  index: number
  ngo: string
}

export const NgoCarouselCard = ({ data, index }: INgoCarouselCardProperties) => {
  return (
    <div className="rounded-[1.25rem] bg-[#f7f7f7] p-3 max-md:bg-white flex flex-col">
      <div className="mt-4 grid grid-cols-2 gap-x-12 gap-y-4 max-md:flex-col max-md:gap-x-4 max-md:gap-y-2">
        <div className="flex flex-col">
          <div className="flex items-center font-bold text-3xl">Stake #{index}</div>
          <div className="flex items-center">Stake Id: {data?.customStakeId}</div>
        </div>
        <Tooltip label={data?.totalDonation.toFixed(18)}>
          <div>
            <CaptionItem
              title="Total donation"
              value={`${formatNumber(data?.totalDonation)} stETH`}
            />
          </div>
        </Tooltip>
        <CaptionItem
          title="Donation percent"
          value={`${formatNumber(data?.percentShare)}%`}
        />
        <Tooltip label={data?.balance.toFixed(18)}>
          <div>
            <CaptionItem
              title="Stake balance"
              value={`${formatNumber(data?.balance)} stETH`}
            />
          </div>
        </Tooltip>
      </div>
    </div>
  )
}
