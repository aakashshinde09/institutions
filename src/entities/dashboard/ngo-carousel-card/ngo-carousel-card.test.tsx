import { expect, test, describe } from 'vitest'
import { render } from '@testing-library/react'
import { NgoCarouselCard } from './ngo-carousel-card.component'
import { Stakes } from '@api/data-contracts'

const mock: Stakes = {
  createdAt: '2024-05-03T15:29:00.000Z',
  customStakeId: 6,
  stackedAmount: 0.000099999999999999,
  percentShare: 1,
  startDate: '2024-05-02T00:00:00.000Z',
  balance: 0.000100009752977152,
  totalDonation: 0,
}

describe('<NgoCarouselCard/>', () => {
  test('Renders correctly', () => {
    const component = render(<NgoCarouselCard ngo='Giga' index={1} data={mock} />)

    expect(component).toMatchSnapshot()
  })
})
