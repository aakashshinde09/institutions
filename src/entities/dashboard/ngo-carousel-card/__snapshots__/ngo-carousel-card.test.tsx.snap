// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<NgoCarouselCard/> > Renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rounded-[1.25rem] bg-[#f7f7f7] p-4 max-md:bg-white"
      >
        <div
          class="mt-5 grid grid-cols-2 gap-x-12 gap-y-4 max-md:mt-4 max-md:flex-col max-md:gap-x-4 max-md:gap-y-2"
        >
          <div
            class="flex flex-col"
          >
            <div
              class="flex items-center font-bold text-3xl"
            >
              Stake #
              1
            </div>
            <div
              class="flex items-center"
            >
              Stake Id: 
              6
            </div>
          </div>
          <div
            class="truncate"
          >
            <p
              class="font-semibold max-md:text-sm max-md:font-normal"
            >
              Total donation
            </p>
            <span
              class="mt-2 text-2xl font-bold max-md:text-base"
            >
              0 stETH
            </span>
          </div>
          <div
            class="truncate"
          >
            <p
              class="font-semibold max-md:text-sm max-md:font-normal"
            >
              Donation percent
            </p>
            <span
              class="mt-2 text-2xl font-bold max-md:text-base"
            >
              1%
            </span>
          </div>
          <div
            class="truncate"
          >
            <p
              class="font-semibold max-md:text-sm max-md:font-normal"
            >
              Stake balance
            </p>
            <span
              class="mt-2 text-2xl font-bold max-md:text-base"
            >
              0.0001 stETH
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rounded-[1.25rem] bg-[#f7f7f7] p-4 max-md:bg-white"
    >
      <div
        class="mt-5 grid grid-cols-2 gap-x-12 gap-y-4 max-md:mt-4 max-md:flex-col max-md:gap-x-4 max-md:gap-y-2"
      >
        <div
          class="flex flex-col"
        >
          <div
            class="flex items-center font-bold text-3xl"
          >
            Stake #
            1
          </div>
          <div
            class="flex items-center"
          >
            Stake Id: 
            6
          </div>
        </div>
        <div
          class="truncate"
        >
          <p
            class="font-semibold max-md:text-sm max-md:font-normal"
          >
            Total donation
          </p>
          <span
            class="mt-2 text-2xl font-bold max-md:text-base"
          >
            0 stETH
          </span>
        </div>
        <div
          class="truncate"
        >
          <p
            class="font-semibold max-md:text-sm max-md:font-normal"
          >
            Donation percent
          </p>
          <span
            class="mt-2 text-2xl font-bold max-md:text-base"
          >
            1%
          </span>
        </div>
        <div
          class="truncate"
        >
          <p
            class="font-semibold max-md:text-sm max-md:font-normal"
          >
            Stake balance
          </p>
          <span
            class="mt-2 text-2xl font-bold max-md:text-base"
          >
            0.0001 stETH
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
